packages/backend/server/src/base/logger/service.ts

```diff
@@ -8,7 +8,7 @@ import { AFFiNELogger } from './logger';
 export const loggerProvider: Provider<LoggerService> = {
   provide: AFFiNELogger,
   useFactory: (config: Config) => {
-    if (config.NODE_ENV !== 'production') {
+    if (config.NODE_ENV !== 'production' || config.isSelfhosted) {
```

Comment:
Instead of a logical judgment, use a configuration of whether or not to enable winston will be better. Each environment has its own configuration, and then add a new deployment environment without fear of missing the logic to come.

```suggestion
    if (!config.affine.enableWinstonLogging) {
```

Reply:
the custom gcloud logger injection should be moved to `plugins/gcloud` module